tools/topology/sof/pipe-passthrough-playback-bt.m4

```diff
@@ -0,0 +1,86 @@
+# Low Latency Passthrough Pipeline and PCM
+#
+# Pipeline Endpoints for connection are :-
+#
+#  host PCM_P --> B0 --> sink DAI0
+
+# Include topology builder
+include(`utils.m4')
+include(`buffer.m4')
+include(`pcm.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`muxdemux.m4')
+include(`bytecontrol.m4')
+
+#
+# Components and Buffers
+#
+
+dnl Configure demux
+dnl name, pipeline_id, routing_matrix_rows
+dnl Diagonal 1's in routing matrix mean that every input channel is
+dnl copied to corresponding output channels in all output streams.
+dnl I.e. row index is the input channel, 1 means it is copied to
+dnl corresponding output channel (column index), 0 means it is discarded.
+dnl There's a separate matrix for all outputs.
+define(matrix1, `ROUTE_MATRIX(PIPELINE_ID,
+			     `BITS_TO_BYTE(1, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 1, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 1 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,1 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,1 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,1 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,1 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,1)')')
```

Comment:
I can see that this might help with upmix/downmix as stated in the commit message.

I don't even know what a demux does on a single pipeline?

Reply:
Demux "let's" us play different channel count that the dai currently has. If mono dai, we'll take only left channel in case of stereo input. Agreed, not very elegant "downmix", but anyway you can play there. mono to mono is just pass through. So now we can pass alsa conformance tests: as we advertise the pipeline for 1-2 channels, you should be able to open/play the device with those channel counts (despite what the DAI config is). If you don't have some kind of upmix/downmix component on the pipeline -> error with mismatching channel count. 

But yeah this was mainly a test for opening/playing the pipeline with different channel count than the dai has. We would really need up/downmix component with automatic configuration or even with some kcontrol.